src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -3228,6 +3228,134 @@ public static void Xml_SerializeClassNestedInStaticClassTest()
+        for (int i = 0; i < band.Instruments.Length; i++)
+        {
+            Assert.Equal(((Brass)band.Instruments.ElementAt(i)).Name, ((Brass)actual.Instruments[i]).Name);
+            Assert.Equal(((Brass)band.Instruments.ElementAt(i)).IsValved, ((Brass)actual.Instruments[i]).IsValved);
+        }
+
+        band = new Orchestra();
+        band.Instruments = new Instrument[1] { new Instrument { Name = "Instrument1" } };
+        attrs = new XmlAttributes();
+        var xArray = new XmlArrayAttribute("CommonInstruments");
+        xArray.Namespace = "http://www.contoso.com";
+        attrs.XmlArray = xArray;
+        attrOverrides = new XmlAttributeOverrides();
+        attrOverrides.Add(typeof(Orchestra), "Instruments", attrs);
+        actual = SerializeAndDeserialize(band,
+@"<?xml version=""1.0""?>
+<Orchestra xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"" xmlns:xsd=""http://www.w3.org/2001/XMLSchema"">
+  <CommonInstruments xmlns=""http://www.contoso.com"">
+    <Instrument>
+      <Name>Instrument1</Name>
+    </Instrument>
+  </CommonInstruments>
+</Orchestra>", () => { return new XmlSerializer(typeof(Orchestra), attrOverrides); });
+        Assert.Equal(band.Instruments.Length, actual.Instruments.Length);
+        for (int i = 0; i < band.Instruments.Length; i++)
+        {
+            Assert.Equal((band.Instruments.ElementAt(i)).Name, (actual.Instruments[i]).Name);
+        }
+
+        band = new Orchestra();
+        var trumpet = new Trumpet() { Name = "TrumpetKeyC", IsValved = false, Modulation = 'C' };
+        band.Instruments = new Instrument[2] { brass, trumpet };
+
+        attrs = new XmlAttributes();
+        var xArrayItem = new XmlArrayItemAttribute(typeof(Brass));
+        xArrayItem.Namespace = "http://www.contoso.com";
+        attrs.XmlArrayItems.Add(xArrayItem);
+        var xArrayItem2 = new XmlArrayItemAttribute(typeof(Trumpet));
+        xArrayItem2.Namespace = "http://www.contoso.com";
+        attrs.XmlArrayItems.Add(xArrayItem2);
+        attrOverrides = new XmlAttributeOverrides();
+        attrOverrides.Add(typeof(Orchestra), "Instruments", attrs);
+        actual = SerializeAndDeserialize(band,
+@"<?xml version=""1.0""?>
+<Orchestra xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"" xmlns:xsd=""http://www.w3.org/2001/XMLSchema"">
+  <Instruments>
+    <Brass xmlns=""http://www.contoso.com"">
+      <Name>Trumpet</Name>
+      <IsValved>true</IsValved>
+    </Brass>
+    <Trumpet xmlns=""http://www.contoso.com"">
+      <Name>TrumpetKeyC</Name>
+      <IsValved>false</IsValved>
+      <Modulation>67</Modulation>
+    </Trumpet>
+  </Instruments>
+</Orchestra>", () => { return new XmlSerializer(typeof(Orchestra), attrOverrides); });
+        Assert.Equal(band.Instruments.Length, actual.Instruments.Length);
+        for (int i = 0; i < band.Instruments.Length; i++)
+        {
+            Assert.Equal((band.Instruments.ElementAt(i)).Name, (actual.Instruments[i]).Name);
```

Comment:
Thanks! I give it a try and it doesn't work, maybe because "band.Instruments" item is a class itself  and the SequenceEqual not support quite well here.